add.go

```diff
@@ -100,6 +106,11 @@ func (b *Builder) Add(destination string, extract bool, source ...string) error
 	if len(source) > 1 && (destfi == nil || !destfi.IsDir()) {
 		return errors.Errorf("destination %q is not a directory", dest)
 	}
+	// Find out which user (and group) should the destination belong to.
```

Comment:
nit of a nit. "should the destination" to "the destination should".  But don't worry about it unless there are other needed changes.